src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderMap.cs

```diff
@@ -47,59 +75,152 @@ private CoseHeaderMap(bool isReadOnly)
+        /// -or
+        /// The encoded bytes contain trailing data or more than one CBOR value.</exception>
         public byte[] GetValueAsBytes(CoseHeaderLabel label) => _headerParameters[label].GetValueAsBytes();
 
+        /// <summary>
+        /// Gets the CBOR encoded value associated with the specified label, as a byte string.
+        /// </summary>
+        /// <param name="label">The label of the value to get.</param>
+        /// <param name="destination">The buffer in which to write the decoded value.</param>
+        /// <returns>The number of bytes written to <paramref name="destination" />.</returns>
+        /// <exception cref="ArgumentException"><paramref name="destination"/> is too small to hold the value.</exception>
+        /// <exception cref="InvalidOperationException">Decodification failed, see the inner exception for the exact exception that was thrown.
+        /// -or
+        /// The encoded bytes contain trailing data or more than one CBOR value.</exception>
         public int GetValueAsBytes(CoseHeaderLabel label, Span<byte> destination) => _headerParameters[label].GetValueAsBytes(destination);
 
+        /// <summary>
+        /// Adds the specified key and value to the dictionary.
+        /// </summary>
+        /// <param name="key">The key of the element to add.</param>
+        /// <param name="value">The value of the element to add.</param>
+        /// <exception cref="InvalidOperationException">The <see cref="CoseHeaderMap"/> is read-only.</exception>
+        /// <exception cref="ArgumentException">The property is set and the encoded bytes in the specified <see cref="CoseHeaderValue"/> contain trailing data or more than one CBOR value.</exception>
         public void Add(CoseHeaderLabel key, CoseHeaderValue value)
         {
             ValidateIsReadOnly();
             ValidateInsertion(key, value);
             _headerParameters.Add(key, value);
         }
 
+        /// <summary>
+        /// Adds the specified value to the <see cref="ICollection{T}"/> with the specified key.
+        /// </summary>
+        /// <param name="item">The <see cref="KeyValuePair{CoseHeaderLabel, CoseHeaderValue}"/> structure representing the key and value to add to the <see cref="CoseHeaderMap"/>.</param>
+        /// <exception cref="InvalidOperationException">The <see cref="CoseHeaderMap"/> is read-only.</exception>
+        /// <exception cref="ArgumentException">The property is set and the encoded bytes in the specified <see cref="CoseHeaderValue"/> contain trailing data or more than one CBOR value.</exception>
         public void Add(KeyValuePair<CoseHeaderLabel, CoseHeaderValue> item) => Add(item.Key, item.Value);
 
+        /// <summary>
+        /// Adds the specified key and value to the <see cref="CoseHeaderMap"/>. The specified value will be encoded as CBOR and stored as a <see cref="CoseHeaderValue"/>.
```

Comment:
I don't think that saying that the value will be encoded is important.  Especially not for the summary (which should ideally only have one sentence).

```suggestion
        /// Adds the specified key and value to the header map.
```